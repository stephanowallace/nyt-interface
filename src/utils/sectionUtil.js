import Constants from 'Constants';

const { SECTION_IDS, SECTIONS } = Constants;

export const orderedSectionsWithIdentifiers = [
  { name: SECTIONS.SCITEC, identifier: SECTION_IDS.SCITEC },
  { name: SECTIONS.HEALTH, identifier: SECTION_IDS.HEALTH },
  { name: SECTIONS.POLITICS, identifier: SECTION_IDS.POLITICS },
  { name: SECTIONS.WORLD, identifier: SECTION_IDS.WORLD },
];

export const getSectionNameById = (sectionId) => {
  const matchingIdSection = orderedSectionsWithIdentifiers
    .filter(section => section.identifier === sectionId);
  return matchingIdSection.length
    ? matchingIdSection[0].name
    : null;
};

export const isSectionIdAvailable = (sectionId) => {
  const availableSectionsIds = orderedSectionsWithIdentifiers
    .map(section => section.identifier);
  return availableSectionsIds.includes(sectionId);
};
