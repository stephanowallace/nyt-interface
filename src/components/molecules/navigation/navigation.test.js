import React from 'react';
import { shallow } from 'enzyme';
import { orderedSectionsWithIdentifiers } from 'Utils/sectionUtil';
import Navigation from './navigation';

describe('Navigation', () => {
  const props = { history: { push: jest.fn() } };

  it('renders default state', () => {
    const component = shallow(<Navigation {...props} />);
    expect(component).toMatchSnapshot();
  });

  it('renders all navigation defined in sectionUtil', () => {
    const component = shallow(<Navigation {...props} />);
    expect(component.find('li')).toHaveLength(orderedSectionsWithIdentifiers.length);
  });

  it('change history path on click', () => {
    const component = shallow(<Navigation {...props} />);
    component.find('.navLink').at(0).simulate('click');
    expect(props.history.push).toHaveBeenCalled();
  });
});
