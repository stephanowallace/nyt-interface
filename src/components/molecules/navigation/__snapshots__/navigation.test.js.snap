// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders default state 1`] = `
<nav>
  <ol
    className="list"
  >
    <li
      key="sciencetechnology"
    >
      <LinkButton
        className="navLink"
        disabled={false}
        onClick={[Function]}
      >
        science & technology
      </LinkButton>
    </li>
    <li
      key="health"
    >
      <LinkButton
        className="navLink"
        disabled={false}
        onClick={[Function]}
      >
        health
      </LinkButton>
    </li>
    <li
      key="politics"
    >
      <LinkButton
        className="navLink"
        disabled={false}
        onClick={[Function]}
      >
        politics
      </LinkButton>
    </li>
    <li
      key="world"
    >
      <LinkButton
        className="navLink"
        disabled={false}
        onClick={[Function]}
      >
        world
      </LinkButton>
    </li>
  </ol>
</nav>
`;
